pkg/reconcile/pipeline/handler/project/impl.go

```diff
@@ -40,6 +40,13 @@ func PreFlightCheck(mandatoryBindingKeys ...string) func(pipeline.Context) {
 					return
 				}
 			}
+		} else {
+			if len(items) == 0 {
+				err := errors.New("no binding item to project")
+				ctx.RetryProcessing(err)
+				ctx.SetCondition(apis.Conditions().NotInjectionReady().Reason(apis.NoBindingItemsReason).Msg(err.Error()).Build())
+				return
```

Comment:
go conventions try to avoid `else` blocks. do this check right after line 31 in a separate if then block.